File: personal-assistant-web/src/api/quickNote.ts
import http from './http';

export interface QuickNote { id: number; content: string; status: string; createdAt: string; updatedAt: string; }
export function createQuickNote(content: string) { return http.post<unknown, { data: number }>('/quick-notes', { content }); }
export function listPendingQuickNotes() { return http.get<unknown, { data: QuickNote[] }>('/quick-notes/pending'); }
export function archiveQuickNote(id: number) { return http.patch(`/quick-notes/${id}/archive`); }

File: personal-assistant-server/src/main/java/com/personal/assistant/module/quicknote/entity/QuickNote.java
package com.personal.assistant.module.quicknote.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 快速记录：任意页面随手记一条，后续可归类到事项/学习/股票等模块。
 */
@Data
@TableName("quick_note")
public class QuickNote {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private String content;

    /**
     * 处理状态：PENDING 待整理 / ARCHIVED 已归档 / CONVERTED 已转为正式记录。
     */
    private String status;

    /**
     * 转化后的目标类型，例如 LIFE / WORK / LEARNING / STOCK，可为空。
     */
    private String convertedType;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}
